pkg/controllers/management/secretmigrator/authconfig.go

```diff
@@ -79,6 +83,45 @@ func (h *handler) migrateAuthConfigToSecret(authConfig *apimgmtv3.AuthConfig, f
 	return updatedAuthConfig, nil
 }
 
+func (h *handler) fixShibbolethSecretReference(obj runtime.Object) (runtime.Object, error) {
+	if apimgmtv3.AuthConfigConditionShibbolethSecretFixed.IsTrue(obj) {
+		return obj, nil
+	}
+
+	shibbConfig := &apimgmtv3.ShibbolethConfig{}
+	updated, err := apimgmtv3.AuthConfigConditionShibbolethSecretFixed.DoUntilTrue(obj, func() (runtime.Object, error) {
+		switch v := obj.(type) {
+		case *apimgmtv3.ShibbolethConfig:
+			shibbConfig = v
+		case *apimgmtv3.AuthConfig:
+			unstructuredConfig, err := getUnstructuredAuthConfigByName(h.authConfigs, v.Name)
+			if err != nil {
+				return nil, err
+			}
+
+			if err := common.Decode(unstructuredConfig.UnstructuredContent(), shibbConfig); err != nil {
+				return nil, fmt.Errorf("unable to decode ShibbolethConfig: %w", err)
+			}
+		}
+
+		if shibbConfig.OpenLdapConfig.ServiceAccountPassword == "cattle-global-data:shibbolethconfig-serviceAccountPassword" {
```

Comment:
Should we `const` this? Just because it's pretty long, and I'm not sure if that is supposed to be the default password?

Reply:
This is there to prevent us modifying the key if it has been modified by a customer.

Reply:
I believe @enrichman's comment is more about creating a go const for the long fixed "cattle-global-data:shibbolethconfig-serviceAccountPassword" string to make the condition not as wide as it currently is.